pkg/gadgets/run/tracer/tracer.go

```diff
@@ -62,6 +63,8 @@ type Config struct {
 }
 
 type Tracer struct {
+	*networktracer.Tracer[types.Event]
```

Comment:
I think we don't should embed this but make it a normal member. I suppose you embedded it to make this tracer implement the `{Attach,Detach}Container` methods, however I think we'll need more control over those methods (like in https://github.com/inspektor-gadget/inspektor-gadget/pull/1866) hence we need to implement those directly in this type. The implementation on this could be a "proxy" and call those methods in `networktracer.Tracer`.